src/qinfer/distributions.py

```diff
@@ -321,6 +322,76 @@ def sample(self, n=1):
         ), len(cumsum_weights) - 1)]
 
     ## MOMENT FUNCTIONS ##
+    @staticmethod
+    def particle_mean(weights, locations):
+        r"""
+        Returns the mean of a function :math:`f` over model
```

Comment:
I agree with @dseuss on this, in that the application to circular means breaks the commonality between ``particle_mean`` and ``est_fn``, unfortunately. I don't think there's an unambiguous sense of how to define a circular mean for arbitrary functions.